addons/missileguidance/functions/fnc_onFiredDeffered.sqf

```diff
@@ -0,0 +1,135 @@
+ *
+ * Return Value:
+ * Args <Array>
+ *
+ * Example:
+ * [player, "", "", "", "ACE_Javelin_FGM148", "", theMissile] call ace_missileguidance_fnc_onFiredDeffered
+ *
+ * Public: No
+ */
+
+params ["_shooter","_weapon","","_mode","_ammo","","_projectile"];
+
+// Bail if guidance is disabled for this ammo
+if ((getNumber (configFile >> "CfgAmmo" >> _ammo >> QUOTE(ADDON) >> "enabled")) != 2) exitWith { ERROR("not enabled=2"); [] };
+
+// MissileGuidance is enabled for this shot
+TRACE_4("enabled",_shooter,_ammo,_projectile,typeOf _shooter);
+
+private _config = configFile >> "CfgAmmo" >> _ammo >> QUOTE(ADDON);
+
+private _target = _shooter getVariable [QGVAR(target), nil];
+private _targetPos = _shooter getVariable [QGVAR(targetPosition), nil];
+private _seekerType = _shooter getVariable [QGVAR(seekerType), nil];
+private _attackProfile = _shooter getVariable [QGVAR(attackProfile), nil];
+if ((getNumber (configFile >> "CfgAmmo" >> _ammo >> QUOTE(ADDON) >> "useModeForAttackProfile")) == 1) then {
+    _attackProfile = getText (configFile >> "CfgWeapons" >> _weapon >> _mode >> QGVAR(attackProfile))
+};
+private _lockMode = _shooter getVariable [QGVAR(lockMode), nil];
+
+private _laserCode = _shooter getVariable [QEGVAR(laser,code), ACE_DEFAULT_LASER_CODE];
+private _laserInfo = [_laserCode, ACE_DEFAULT_LASER_WAVELENGTH, ACE_DEFAULT_LASER_WAVELENGTH];
+
+TRACE_6("getVars",_target,_targetPos,_seekerType,_attackProfile,_lockMode,_laserCode);
+
+private _launchPos = getPosASL (vehicle _shooter);
+
+if (isNil "_seekerType" || {!(_seekerType in (getArray (_config >> "seekerTypes")))}) then {
+    _seekerType = getText (_config >> "defaultSeekerType");
+};
+if (isNil "_attackProfile" || {!(_attackProfile in (getArray (_config >> "attackProfiles")))}) then {
+    _attackProfile = getText (_config >> "defaultAttackProfile");
+};
+if (isNil "_lockMode" || {!(_lockMode in (getArray (_config >> "seekerLockModes")))}) then {
+    _lockMode = getText (_config >> "defaultSeekerLockMode");
+};
+
+// If we didn't get a target, try to fall back on tab locking
+if (isNil "_target") then {
+    if (!isPlayer _shooter) then {
+        // This was an AI shot, lets still guide it on the AI target
+        _target = _shooter getVariable [QGVAR(vanilla_target), nil];
+        TRACE_1("Detected AI Shooter!", _target);
+    } else {
+        private _canUseLock = getNumber (_config >> "canVanillaLock");
+        // @TODO: Get vanilla target
+        if (_canUseLock > 0 || difficulty < 1) then {
+            private _vanillaTarget = cursorTarget;
+
+            TRACE_1("Using Vanilla Locking", _vanillaTarget);
+            if (!isNil "_vanillaTarget") then {
+                _target = _vanillaTarget;
+            };
+        };
+    };
+};
+
+// Array for seek last target position
+private _seekLastTargetPos = (getNumber ( _config >> "seekLastTargetPos")) == 1;
+private _lastKnownPosState = [_seekLastTargetPos];
+if (_seekLastTargetPos && {!isNil "_target"}) then {
```

Comment:
Surely the lazy eval + nil check is more expensive than the nil check alone ? The check is on a separate var as well.
```suggestion
if (_seekLastTargetPos && !isNil "_target") then {
```